### PR TITLE
Use py.test test runner

### LINKED ISSUE #227: Switch tests to py.test
@hynek loves pytest.

### PR DESCRIPTION
**Pull request by hynek**
Fixes #227

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -12,5 +13,5 @@ setenv =
 # The standard library unittest module can run tests on Python 2.7 and newer
 commands =
     python -c "import OpenSSL.SSL; print(OpenSSL.SSL.SSLeay_version(OpenSSL.SSL.SSLEAY_VERSION))"
-    coverage run --branch --source=OpenSSL setup.py test
+    coverage run --branch --source=OpenSSL {envdir}/bin/py.test OpenSSL
```
Comment (alex) [inline tox.ini]: You can use `-m pytest` instead of the `{envdir}` thing

File: tox.ini [hunk 2]
```diff
@@ -12,5 +13,5 @@ setenv =
 # The standard library unittest module can run tests on Python 2.7 and newer
```
Comment (exarkun) [inline tox.ini]: This comment is being out-dated, it appears.

File: tox.ini [hunk 3]
```diff
@@ -12,5 +13,5 @@ setenv =
...
 commands =
     python -c "import OpenSSL.SSL; print(OpenSSL.SSL.SSLeay_version(OpenSSL.SSL.SSLEAY_VERSION))"
-    coverage run --branch --source=OpenSSL setup.py test
+    coverage run --branch --source=OpenSSL -m pytest OpenSSL
```
Comment (exarkun) [inline tox.ini]: Do you want to leave the test suite information in `setup.py` even though it's no longer being used by the CI system?

Comment (hynek) [inline tox.ini]: I will make `setup.py test` work again once I’ve verified that the test suite actually works.

**Pull request merged**